index.js

```diff
@@ -1,39 +1,171 @@
-    readlineInterface.question(questionText, resolve);
-  });
+	return new Promise((resolve, reject) => {
+		readlineInterface.question(questionText, resolve);
+	});
 }
 
 // remember the StateMachine lecture
 // https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
+let states = { // I'm sure I need functions inside this for change the inventory of the room?
+	"182main": {
+		canChangeTo: ["foyer", "mr.mikes", "muddys"],
+		name: "182 Main St.",
+		welcomeMessage: `
+182 Main St. 
+You are standing on Main Street between Church and South Winooski.
+There is a door here. A keypad sits on the handle.
+On the door is a handwritten sign.
+
+`,
+		readSign: `
+
+The sign says "Welcome to Burlington Code Academy! Come on 
+up to the third floor. If the door is locked, use the code
+12345."
+
+`
+	},
+	"foyer": {
+		canChangeTo: ["182main", "stairs"],
+		name: "Foyer",
+		welcomeMessage: `
+You are in a foyer. Or maybe it's an antechamber. Or a 
+vestibule. Or an entryway. Or an atrium. Or a narthex.
+But let's forget all that fancy flatlander vocabulary,
+and just call it a foyer. In Vermont, this is pronounced
+"FO-ee-yurr".
+
+A copy of Seven Days lies in a corner.
+
+`,
+		readPaper: `
+You pick up the paper and leaf through it looking for comics 
+and ignoring the articles, just like everybody else does.
+
+`,
+		takePaper: `
+You are carrying:
+A copy of Seven Days, Vermont's Alt-Weekly
+
+`,
+		dropPaper: `
+Paper has been dropped.
+
+`
+	},
+	"stairs": { canChangeTo: ["classroom", "foyer"], name: "Stairs" },
+	"mr.mikes": { canChangeTo: ["182main"] },
+	"muddys": { canChangeTo: ["182main"] },
+	"classroom": { canChangeTo: ["stairs"],
+		welcomeMessage: `
+		
+You made it to class but no one has coffee including yourself. 
+So program explodes enter exit and leave program now so you can fix it!
+
+` }
 };
 
-let currentState = "green";
+/*let character = { Character Object. Still not sure exactly how I want to implement.
+	"inventory": {
+
+	},
+	"name": 'agent Z',
+	"appetite": {},
+
+}*/ 
+
+let currentState = "182main";
+
+function enterState(newState) {//  State Machine for room transition rules. 
+let validTransitions = states[currentState].canChangeTo; //Do I need another seperate function for inventory?
+	if (validTransitions.includes(newState)) {
+		currentState = newState;
+	} else {
+		throw "Invalid state transition attempted - from " +
+			currentState +
+			" to " +
+			newState;
+	}
+}
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
-  } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
-  }
+function checkPinCode(code) {
+	code = answer.includes("enter code", 0);
+	if (code === "true") {
```

Comment:
I believe you're going to actually want to be checking against the boolean `true` rather than the string `'true'` Overall I really like this way of validating the door code.

Reply:
Thank you! I was sure whether you need the string or if true would work. I never ended up implementing this into my code but wanted to.

Reply:
If code is a boolean you could even just say `if (code) {...}` since that will evaluate to true or false all by itself.